lib/cocoapods/installer/xcode/pods_project_generator/project_generator.rb

```diff
@@ -38,6 +38,11 @@ class ProjectGenerator
         #
         attr_reader :podfile_path
 
+        # @return [Bool] Bool indicating if this project is a pod target subproject.
+        # Used by `generate_multiple_pod_projects` installation option.
+        #
+        attr_reader :pod_target_subproject
```

Comment:
is there a way to kill this flag entirely and have things scope down automatically just by passing the project?

Reply:
I don't believe so since this class is responsible for creating/preparing `Pod::Project` which also requires the flag. One may to mask the flag might be subclassing this project generator and creating something like a `SubprojectGenerator` object.

Reply:
I prefer the flag than subclassing :)